src/test/scala/com/softwaremill/realworld/articles/ArticlesEndpointsSpec.scala

```diff
@@ -1,27 +1,138 @@
+                "How to train your dragon",
+                "Ever wonder how?",
+                "It takes a Jacobian",
+                Nil,
+                Instant.ofEpochMilli(1455765776637L),
+                Instant.ofEpochMilli(1455767315824L),
+                false,
+                0,
+                ArticleAuthor("jake", "I work at statefarm", "https://i.stack.imgur.com/xHWG8.jpg", following = false)
+              )
+            )
+            && contains(
+              Article(
+                "how-to-train-your-dragon-2",
+                "How to train your dragon 2",
+                "So toothless",
+                "Its a dragon",
+                Nil,
+                Instant.ofEpochMilli(1455765776637L),
+                Instant.ofEpochMilli(1455767315824L),
+                false,
+                0,
+                ArticleAuthor("jake", "I work at statefarm", "https://i.stack.imgur.com/xHWG8.jpg", following = false)
+              )
+            )
+        )
+      )
+    } @@ TestAspect.before(withFixture("fixtures/articles/original-spec-data.sql"))
+      @@ TestAspect.after(clearDb),
+    test("try getting not existing article") {
+      val backendStub = TapirStubInterpreter(SttpBackendStub(new RIOMonadError[ArticlesService with DataSource]))
+        .whenServerEndpoint(ArticlesEndpoints.get)
+        .thenRunLogic()
+        .backend()
+
+      val response = basicRequest
+        .get(uri"http://test.com/api/articles/unknown-article")
+        .response(asJson[Article])
+        .send(backendStub)
+
+      assertZIO(response.map(_.body))(
+        isLeft(equalTo(HttpError("Article with slug unknown-article doesn't exist.", sttp.model.StatusCode(400))))
+      )
+
+    } @@ TestAspect.before(withFixture("fixtures/articles/original-spec-data.sql"))
+      @@ TestAspect.after(clearDb),
+    test("get existing article") {
+      val backendStub = TapirStubInterpreter(SttpBackendStub(new RIOMonadError[ArticlesService with DataSource]))
+        .whenServerEndpoint(ArticlesEndpoints.get)
+        .thenRunLogic()
+        .backend()
+
+      val response = basicRequest
+        .get(uri"http://test.com/api/articles/how-to-train-your-dragon-2")
+        .response(asJson[Article])
+        .send(backendStub)
+
+      assertZIO(response.map(_.body))(
+        isRight(
+          equalTo(
+            Article(
+              "how-to-train-your-dragon-2",
+              "How to train your dragon 2",
+              "So toothless",
+              "Its a dragon",
+              Nil,
+              Instant.ofEpochMilli(1455765776637L),
+              Instant.ofEpochMilli(1455767315824L),
+              false,
+              0,
+              ArticleAuthor("jake", "I work at statefarm", "https://i.stack.imgur.com/xHWG8.jpg", following = false)
+            )
+          )
+        )
+      )
+    } @@ TestAspect.before(withFixture("fixtures/articles/original-spec-data.sql"))
+      @@ TestAspect.after(clearDb)
```

Comment:
can we somehow extract these aspects so that they apply to all tests, to avoid repetition?

Reply:
I could put them in suite level, but one test has different `before` => https://github.com/softwaremill/realworld-tapir-zio/blob/5769596ed33d6d2ddda1cf3e9a50f94e1eeaf249/src/test/scala/com/softwaremill/realworld/articles/ArticlesEndpointsSpec.scala#L41. That's why they are on every test.

Reply:
I've created two suites (with empty and populated db), so there is a little less repetition.